src/ast.rs

```diff
@@ -466,6 +467,10 @@ where
     // The original input string.
     input: &'a [u8],
 
+    // `Identifier`s will be placed here, so `UnnamedTypeName` can utilize and print
+    // out Constructor/Destructor used.
+    source_name: Option<String>,
```

Comment:
Is there a way to avoid a heap allocation here and use a `&'a str` instead of a `String`?

Reply:
I can definitely do that! :smile: